pkg/services/tree/service.go

```diff
@@ -266,11 +310,22 @@ func getCID(rawCID []byte) *cidSDK.ID {
 	return cidSDK.NewFromV2(&cidV2)
 }
 
-func constructMeta(arr []*KeyValue) []pilorama.KeyValue {
+func protoToMeta(arr []*KeyValue) []pilorama.KeyValue {
```

Comment:
Can `arr` contain `nil`?

Reply:
`proto.Unmarshal` returns slice with empty structs, but added a check, just in case.